src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpFormatting.cs

```diff
@@ -0,0 +1,214 @@
+        public void AlignOpenBraceWithMethodDeclaration()
+        {
+            SetUpEditor(@"
+$$class C
+{
+    void Main()
+     {
+    }
+}");
+
+            Editor.FormatDocument();
+            VerifyTextContains(@"
+class C
+{
+    void Main()
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void FormatOnSemicolon()
+        {
+            SetUpEditor(@"
+public class C
+{
+    void Foo()
+    {
+        var x =        from a             in       new List<int>()
+    where x % 2 = 0
+                      select x   ;$$
+    }
+}");
+
+            SendKeys(VirtualKey.Backspace, ";");
+            VerifyTextContains(@"
+public class C
+{
+    void Foo()
+    {
+        var x = from a in new List<int>()
+                where x % 2 = 0
+                select x;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void FormatSelection()
+        {
+            SetUpEditor(@"
+public class C {
+    public void M( ) {$$
+    }
+}");
+
+            SelectTextInCurrentDocument("public void M( ) {");
+            Editor.FormatSelection();
+            VerifyTextContains(@"
+public class C {
+    public void M()
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void PasteCodeWithLambdaBody1()
+        {
+            SetUpEditor(@"
+using System;
+class Program
+{
+    static void Main()
+    {
+        Action a = () =>
+        {
+            using (null)
+            {
+                $$
+            }
+        };
+    }
+}");
+            Editor.Paste(@"        Action b = () =>
+        {
+
+        };");
+            VerifyTextContains(@"
+using System;
+class Program
+{
+    static void Main()
+    {
+        Action a = () =>
+        {
+            using (null)
+            {
+                Action b = () =>
+                {
+
+                };
+            }
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void PasteCodeWithLambdaBody2()
+        {
+            SetUpEditor(@"
+using System;
+class Program
+{
+    static void Main()
+    {
+        Action a = () =>
+        {
+            using (null)
+            {
+                $$
+            }
+        };
+    }
+}");
+            Editor.Paste(@"        Action<int> b = n =>
+        {
+            Console.Writeline(n);
+        };");
+            VerifyTextContains(@"
+using System;
+class Program
+{
+    static void Main()
+    {
+        Action a = () =>
+        {
+            using (null)
+            {
+                Action<int> b = n =>
+                {
+                    Console.Writeline(n);
+                };
+            }
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void PasteCodeWithLambdaBody3()
```

Comment:
Are the members of the PasteCodeWithLambdaBodyN family testing the integration in different ways? (IE, could we just have one of these and cover the rest with unit tests?)